### PR TITLE
Sprint 6 solution

### PR DESCRIPTION
**Pull request by mat1x228**

### CODE REVIEW

**Comment by mat1x228** [comment]
Случайно открыл новый ПР 
Не понимаю почему тесты не проходят 
Check: ./src/interfaces/HistoryManager.java
Starting audit...
Audit done.
Check: ./src/interfaces/TaskManager.java
Starting audit...
Error:  /home/runner/work/java-kanban/java-kanban/./src/interfaces/TaskManager.java:7:8: Unused import - java.util.ArrayList. [UnusedImports]
Error:  /home/runner/work/java-kanban/java-kanban/./src/interfaces/TaskManager.java:13:5: 'METHOD_DEF' should be separated from previous line. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 2 errors.
Error: Process completed with exit code 2.

**Comment by kvach2avl** [comment]
> Не понимаю почему тесты не проходят Check: ./src/interfaces/HistoryManager.java Starting audit... Audit done. Check: ./src/interfaces/TaskManager.java Starting audit... Error: /home/runner/work/java-kanban/java-kanban/./src/interfaces/TaskManager.java:7:8: Unused import - java.util.ArrayList. [UnusedImports] Error: /home/runner/work/java-kanban/java-kanban/./src/interfaces/TaskManager.java:13:5: 'METHOD_DEF' should be separated from previous line. [EmptyLineSeparator]

Так там же прямо причины написаны с номерами строк. В первом случае неиспользуемый импорт, во втором случае не хватает пустой строки между описаниями методов.

File: src/managerLogic/InMemoryTaskManager.java
Comment (kvach2avl) [inline src/managerLogic/InMemoryTaskManager.java:1]: В методах removeEpicById(), clearSubtasks(), clearEpics() и clearTasks() тоже происходит удаление (под)задач/эпиков и, значит, должна очищаться соответствующая история

File: src/interfaces/HistoryManager.java
Comment (kvach2avl) [inline src/interfaces/HistoryManager.java:1]: Метод clearHistory() по ТЗ не требуется

**Review by kvach2avl — COMMENTED** [review]
На будущее просьба не открывать новых ПР - это лишает возможности комментировать код.
По тестам ответил в комментарии, остальные замечания - в комментарии к классу.

File: src/interfaces/HistoryManager.java
Comment (mat1x228) [reply src/interfaces/HistoryManager.java:1]: убрал

File: src/managerLogic/InMemoryTaskManager.java
Comment (mat1x228) [reply src/managerLogic/InMemoryTaskManager.java:1]: поправил

**Comment by mat1x228** [comment]
Не видит импорты для тестов, пытался решить проблему в зависимостях в проекте, добавил, не помогло

Run find ./src -type f -name "*.java" > sources.txt
./src/tests/TaskTest.java:2: error: package org.junit.jupiter.api does not exist
import static org.junit.jupiter.api.Assertions.*;
                                   ^
./src/tests/TaskTest.java:6: error: package org.junit.jupiter.api does not exist
import org.junit.jupiter.api.BeforeAll;
                            ^
./src/tests/TaskTest.java:7: error: package org.junit.jupiter.api does not exist
import org.junit.jupiter.api.Test;
                            ^
./src/tests/manager/InMemoryHistoryManagerTest.java:5: error: package org.junit.jupiter.api does not exist
import org.junit.jupiter.api.Test;
                            ^
./src/tests/manager/InMemoryHistoryManagerTest.java:11: error: package org.junit.jupiter.api does not exist
import static org.junit.jupiter.api.Assertions.assertEquals;
                                   ^
./src/tests/manager/InMemoryHistoryManagerTest.java:11: error: static import only from classes and interfaces
import static org.junit.jupiter.api.Assertions.assertEquals;
^
./src/tests/manager/InMemoryTaskManagerTest.java:7: error: package org.junit does not exist
import org.junit.Test;
                ^
./src/tests/manager/InMemoryTaskManagerTest.java:13: error: package org.junit.jupiter.api does not exist
import static org.junit.jupiter.api.Assertions.*;
                                   ^
./src/tests/manager/ManagerTest.java:6: error: package org.junit.jupiter.api does not exist
import org.junit.jupiter.api.Test;
                            ^
./src/tests/manager/ManagerTest.java:[8](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:9): error: package org.junit.jupiter.api does not exist
import static org.junit.jupiter.api.Assertions.assertNotNull;
                                   ^
./src/tests/manager/ManagerTest.java:8: error: static import only from classes and interfaces
import static org.junit.jupiter.api.Assertions.assertNotNull;
^
./src/tests/TaskTest.java:14: error: cannot find symbol
    @BeforeAll
     ^
  symbol:   class BeforeAll
  location: class TaskTest
./src/tests/TaskTest.java:1[9](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:10): error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class TaskTest
./src/tests/TaskTest.java:26: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class TaskTest
./src/tests/TaskTest.java:33: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class TaskTest
./src/tests/manager/InMemoryHistoryManagerTest.java:[15](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:16): error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class InMemoryHistoryManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:[17](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:18): error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:43: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:60: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class InMemoryTaskManagerTest
./src/tests/manager/ManagerTest.java:15: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class ManagerTest
./src/tests/manager/ManagerTest.java:[20](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:21): error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class ManagerTest
./src/tests/TaskTest.java:23: error: cannot find symbol
        assertEquals(task1, task2);
        ^
  symbol:   method assertEquals(Task,Task)
  location: class TaskTest
./src/tests/TaskTest.java:30: error: cannot find symbol
        assertEquals(epic1, epic2);
        ^
  symbol:   method assertEquals(Epic,Epic)
  location: class TaskTest
./src/tests/TaskTest.java:38: error: cannot find symbol
        assertEquals(subtask1, subtask2);
        ^
  symbol:   method assertEquals(SubTask,SubTask)
  location: class TaskTest
./src/tests/manager/InMemoryHistoryManagerTest.java:25: error: cannot find symbol
        assertEquals(1, history.size());
        ^
  symbol:   method assertEquals(int,int)
  location: class InMemoryHistoryManagerTest
./src/tests/manager/InMemoryHistoryManagerTest.java:26: error: cannot find symbol
        assertEquals(task1, history.get(0));
        ^
  symbol:   method assertEquals(Task,Task)
  location: class InMemoryHistoryManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:35: error: cannot find symbol
        assertNotNull(foundTask);
        ^
  symbol:   method assertNotNull(Task)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:36: error: cannot find symbol
        assertNotNull(foundEpic);
        ^
  symbol:   method assertNotNull(Epic)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:37: error: cannot find symbol
        assertNotNull(foundSubTask);
        ^
  symbol:   method assertNotNull(SubTask)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:38: error: cannot find symbol
        assertEquals(task.getDiscr(), foundTask.getDiscr());
        ^
  symbol:   method assertEquals(String,String)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:39: error: cannot find symbol
        assertEquals(epic.getDiscr(), foundEpic.getDiscr());
        ^
  symbol:   method assertEquals(String,String)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:40: error: cannot find symbol
        assertEquals(subTask.getDiscr(), foundSubTask.getDiscr());
        ^
  symbol:   method assertEquals(String,String)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:57: error: cannot find symbol
        assertNotEquals(epic.getId(), generatedTask.getId());
        ^
  symbol:   method assertNotEquals(int,int)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:70: error: cannot find symbol
        assertEquals(originalTask.getName(), clonedTask.getName());
        ^
  symbol:   method assertEquals(String,String)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:71: error: cannot find symbol
        assertEquals(originalTask.getDiscr(), clonedTask.getDiscr());
        ^
  symbol:   method assertEquals(String,String)
  location: class InMemoryTaskManagerTest
./src/tests/manager/InMemoryTaskManagerTest.java:72: error: cannot find symbol
        assertEquals(originalTask.getStatus(), clonedTask.getStatus());
        ^
  symbol:   method assertEquals(Progress,Progress)
  location: class InMemoryTaskManagerTest
./src/tests/manager/ManagerTest.java:17: error: cannot find symbol
        assertNotNull(taskManager);
        ^
  symbol:   method assertNotNull(TaskManager)
  location: class ManagerTest
./src/tests/manager/ManagerTest.java:[22](https://github.com/mat1x228/java-kanban/actions/runs/8605736845/job/23582628973?pr=3#step:6:23): error: cannot find symbol
        assertNotNull(historyManager);
        ^
  symbol:   method assertNotNull(HistoryManager)
  location: class ManagerTest
38 errors

**Review by kvach2avl — COMMENTED** [review]
К основному коду вопросов нет, однако обязательно нужно добиться "зеленой" сборки в части автотестов.
(хинт: судя по жалобам речь идёт о невозможности найти библиотеку для тестирования)
Рекомендация обратиться к наставнику остаётся в силе.

**Comment by mat1x228** [comment]
Пофиксил ошибки при сборке

**Review by kvach2avl — COMMENTED** [review]
Ок.
На будущее: файлы настроек IDE (типа codeStyleConfig.xml, java-kanban.iml и т.д.) и результаты компиляции (*.class, *.jar и т.д.) в репозиторий добавлять не нужно. При командной разработке на этих файлах всегда будут конфликты, поэтому они должны быть исключены из коммитов

**Pull request merged**